### PR TITLE
Conflict resolution

### LINKED ISSUE #456: Conflict resolution component
This issue handles the creation of the conflict resolution component.

### PR DESCRIPTION
**Pull request by jitsedesmet**
This PR closes  #456.

### CODE REVIEW

File: frontend/src/api/calls/AssignmentCalls.ts [hunk 1]
```diff
@@ -1,21 +1,36 @@
...
 /**
- * Fetches all assignments on a given AssignmentLinksUrl
+ * Gets all [IAssignment] entities on an url hosting [IPage].
+ * @param url url hosting the IPage
  */
-export function getAllAssignmentsFormLinks(url: string): Promise<IAssignment[]> {
-    return <Promise<IAssignment[]>>getAllEntitiesFromLinksUrl(url, assignmentCollectionName);
+export async function getAllAssignmentsFromPage(url: string): Promise<IAssignment[]> {
+    const res = await getAllEntitiesFromPage(url, assignmentCollectionName);
+    console.log(res);
```
Comment (ruvmello) [inline frontend/src/api/calls/AssignmentCalls.ts:24]: can go

File: frontend/src/api/calls/AssignmentCalls.ts [hunk 2]
```diff
@@ -52,6 +67,40 @@ export async function deleteAssignment(
...
+export async function invalidateAssignment(assignmentUrl: string): Promise<IAssignment | undefined> {
+    const response = await basePatch(assignmentUrl, { isValid: false });
+    const assignment = <IAssignment>response.data;
+    try {
+        // Check if response is a valid assignment.
+        assignment._links.self.href;
+        return assignment;
+    } catch (e) {
+        return undefined;
+    }
```
Comment (ruvmello) [inline frontend/src/api/calls/AssignmentCalls.ts:83]: Seems weird to use a try and catch here, can't you just look if assignment is defined? or of assignment._links is defined or something, as this is pretty ugly.

File: frontend/src/components/projectAssignment/assignmentStudentRow.tsx [hunk 3]
```diff
@@ -9,26 +9,18 @@ import { emptyStudent, IStudent } from "../../api/entities/StudentEntity";
...
 }
 
-export default function AssignmentStudentRow({
-    assignment,
-    removeCallback,
-    registerSortKey,
-}: IAssignmentStudentProps) {
+export default function AssignmentStudentRow({ assignment, removeCallback }: IAssignmentStudentProps) {
     const { t } = useTranslation("common");
+    console.log(assignment._links.student.href);
```
Comment (ruvmello) [inline frontend/src/components/projectAssignment/assignmentStudentRow.tsx:23]: can go

File: frontend/src/components/conflictResolution/conflictResolutionItem.tsx [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    assignment,
+}: ProjectSkillRegisterProps) {
+    const { data: receivedSkill, error: skillError } = useSWR(
+        assignment._links.projectSkill.href,
+        getProjectSkillOnUrl
+    );
+
+    const receivedProjectSkillUrl = receivedSkill?._links.self.href;
+    const assignmentUrl = assignment._links.self.href;
+    const hasErrored = !!skillError;
```
Comment (ruvmello) [inline frontend/src/components/conflictResolution/conflictResolutionItem.tsx:43]: double exclamation mark?

File: frontend/src/components/conflictResolution/conflictResolutionItem.tsx [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+
+    useEffect(() => {
+        if (receivedProjectSkillUrl) {
+            registerAssignment(receivedProjectSkillUrl, assignmentUrl);
+        } else if (hasErrored) {
+            // Let us hope this doesn't happen too often
+            removeAssignment(assignmentUrl);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (ruvmello) [inline frontend/src/components/conflictResolution/conflictResolutionItem.tsx:52]: Should this be disabled?

**Review by ruvmello — CHANGES_REQUESTED** [review]
the conflictresolutionitem file is pretty big and I don't fully understand everything from just looking at it in github. I read you will refactor it, so for know i'll just trust you that it works for that file.

File: frontend/src/components/conflictResolution/conflictResolutionItem.tsx (hunk 4 above)
Comment (jitsedesmet) [reply frontend/src/components/conflictResolution/conflictResolutionItem.tsx:43]: Yeah, Cast to boolean.. weird... I know. but intellij wants this.

File: frontend/src/components/conflictResolution/conflictResolutionItem.tsx (hunk 5 above)
Comment (jitsedesmet) [reply frontend/src/components/conflictResolution/conflictResolutionItem.tsx:52]: Yeah, the hooks are dependencies are configured correctly so this should be allowed.

**Comment by jitsedesmet** [comment]
This PR now also contains tests! Yey (thanks to my sleep disorder)

**Review by ruvmello — APPROVED** [review]

**Pull request merged**